src/http/client.cc

```diff
@@ -390,7 +390,20 @@ future<> client::do_make_request(connection& con, request& req, reply_handler& h
             });
         }
 
-        return handle(rep, con.in(rep)).finally([reply = std::move(reply)] {});
+        return handle(rep, con.in(rep)).then([reply = std::move(reply), &con] {
+            if (reply->content_length > reply->consumed_content) {
+                auto bytes_left = reply->content_length - reply->consumed_content;
+                if (bytes_left <= 128 * 1024) {
```

Comment:
This should be at least configurable client variable. not a magic constant in the middle of the code